docs/rfcs/1012-parsec-auth.md

```diff
@@ -0,0 +1,430 @@
+
+  For the FIDO2 case the included challenge is used as is since the private key is not shared with the server.
+
+- An asymmetric key pair (`AUTH_MEDIUM_PUB_KEY`, `AUTH_MEDIUM_PRIV_KEY`)
+
+  They're generated using a cryptographic secure pseudo-random number generator (CSPRNG).
+
+  > This step is not needed when using a fido2 device since, it provides the asymmetric keys securely.
+
+- A symmetric key used to encrypt `AUTH_MEDIUM_PRIV_KEY` (`AUTH_MEDIUM_SYM_KEY`) alongside its parameters used to generate it `AUTH_MEDIUM_SYM_KEY_PARAMS`.
+
+  When using a password, it's generated from the function `PBKDF_B`
+
+  For FIDO2, we can bypass that step since we do not share `AUTH_MEDIUM_PRIV_KEY` since it's stored on the device.
+
+- A randomly generated symmetric key `ACCOUNT_MANIFEST_SYM_KEY`, it will be used to encrypt the account manifest.
+
+- Generate the account manifest `ACCOUNT_MANIFEST`,
+
+  the manifest will store:
+
+  - The list of authentication public keys
+  - The list of keys used to encrypt the devices (the list is empty at the beginning)
+
+  > The manifest will be encrypted to prevent the server to temper with it.
+
+> [!CAUTION]
+> The output of `PBKDF_A` and `PBKDF_B` should be different to avoid leaking the private key.
+
+
+```mermaid
+sequenceDiagram
+  autonumber
+  participant Alice
+  participant S as Authentification server
+
+  Alice ->> S: Provide its email for the account creation
+  S ->> Alice: Send an email with a unique code need to continue the process
+  alt Choose between password or fido2
+    Alice ->> Alice: Choose a password
+    Alice ->> Alice: Derive `AUTH_MEDIUM_HMAC_KEY` using the password and `PBKDF_A`
+    Alice ->> Alice: Create `AUTH_MEDIUM_SYM_KEY` using the password and `PBKDF_B`
+    Alice ->> Alice: Generate `AUTH_MEDIUM_PRIV_KEY`, `AUTH_MEDIUM_PUB_KEY`
+    Alice ->> Alice: Encrypt `AUTH_MEDIUM_PRIV_KEY` with `AUTH_MEDIUM_SYM_KEY`
+  else
+    Alice ->> Alice: Generate `AUTH_MEDIUM_HMAC_KEY` using the fido2 device.
+    Alice ->> Alice: Use the public key of the FIDO2 device as `AUTH_MEDIUM_PUB_KEY`, the private key of FIDO2 device as `AUTH_MEDIUM_PRIV_KEY`
+  end
+
+  Alice ->> Alice: Create a bundle containing `AUTH_MEDIUM_PUB_KEY` (`ACCOUNT_MANIFEST`)
+  Alice ->> Alice: Create a symmetric key `ACCOUNT_MANIFEST_SYM_KEY`
+  Alice ->> Alice: Encrypt the bundle `ACCOUNT_MANIFEST` with `ACCOUNT_MANIFEST_SYM_KEY` (`enc(ACCOUNT_MANIFEST)`)
+  Alice ->> Alice: Encrypt `ACCOUNT_MANIFEST_SYM_KEY` with `AUTH_MEDIUM_PUB_KEY` (`enc(ACCOUNT_MANIFEST_SYM_KEY)`)
+  Alice ->> S: Send the require data to finish the account creation (`enc(AUTH_MEDIUM_PRIV_KEY)`, `AUTH_MEDIUM_SYM_KEY_PARAMS`, `enc(ACCOUNT_MANIFEST)`, `enc(ACCOUNT_MANIFEST_SYM_KEY)`, `AUTH_MEDIUM_HMAC_KEY`, email validation token)
+```
+
+At the end of the process, the server should save the following information:
+
+- An entry for the user account:
+
+  - A UUID (`user_id`)
```

Comment:
nit: maybe it could be renamed to account_id to avoid confusion.

Reply:
done